autosklearn/data/validation.py

```diff
@@ -533,3 +538,34 @@ def join_and_check(
                              "Pandas DataFrames, numpy arrays, scipy csr  or  python lists. "
                              "Kindly cast your targets to a supported type."
                              )
+
+    def _is_data_valid_for_classification(self, y: np.ndarray) -> None:
+        """
+        Makes sure that the data provided for classification is
+        not continuous
+
+        This check can only catch for undesired numerical data (float),
+        as object dtype requires further processing. For instance, if the user
+        provided a '1.23', it will be interpreted as a valid categorical value
+        valid to encode.
+        """
+
+        if hasattr(y, "iloc"):
+            # Get only the numerical columns
+            y_num = y._get_numeric_data()
+
+            if not y_num.empty and np.any(y_num.to_numpy() != y_num.to_numpy().astype(int)):
+                raise ValueError("classification with data of type "
+                                 "continuous is not supported. The columns {} "
+                                 "where identified as continuous.".format(
+                                     y_num.columns
+                                    )
+                                 )
+        elif not scipy.sparse.issparse(y):
+            # It should be a list or numpy array
+            # if this point was reached
+            y_array = np.array(y)
```

Comment:
Could you please check if `y` is already a numpy array and do not make a copy if that's the case?